### PR TITLE
feat: Select all projects checkbox

### PR DESCRIPTION
**Pull request by Skn0tt**
Works towards https://github.com/microsoft/playwright/issues/33865. Once we have a "unselect all" button, we can automatically select all projects by default.

https://github.com/user-attachments/assets/e542a4d7-f71e-442c-bb15-5ed84b73f3b4

### CODE REVIEW

File: src/settingsView.ts [hunk 1]
```diff
@@ -316,12 +319,34 @@ function htmlForWebview(vscode: vscodeTypes.VSCode, extensionUri: vscodeTypes.Ur
           div.appendChild(label);
           projectsElement.appendChild(div);
         }
+
+        const allEnabled = projects.every(p => p.enabled);
```
Comment (agg23) [inline src/settingsView.ts:323]: You set this to a variable, then never use it, and redo the same calculation in the next line.

File: src/settingsView.ts [hunk 2]
```diff
@@ -316,12 +319,34 @@ function htmlForWebview(vscode: vscodeTypes.VSCode, extensionUri: vscodeTypes.Ur
...
+          checkbox.dispatchEvent(new Event('change'));
+        }
+      });
+
       const vscode = acquireVsCodeApi();
       for (const input of document.querySelectorAll('input[type=checkbox]')) {
         input.addEventListener('change', event => {
-          vscode.postMessage({ method: 'toggle', params: { setting: event.target.getAttribute('setting') } });
+          const setting = event.target.getAttribute('setting');
+          if (setting)
```
Comment (agg23) [inline src/settingsView.ts:348]: What happens if there's no setting (I don't know this VSCode API)?

**Review by agg23 — APPROVED** [review]

File: src/settingsView.ts (hunk 1 above)
Comment (Skn0tt) [reply src/settingsView.ts:323]: oops!

File: src/settingsView.ts (hunk 2 above)
Comment (Skn0tt) [reply src/settingsView.ts:348]: It's standard DOM. We have some input elements in the HTML that have a setting attribute, e.g. this:

https://github.com/Skn0tt/playwright-vscode/blob/9f109ef3ce9af487b00136196fc67810cb0e1ee2/src/settingsView.ts#L276

Before this change, there were no elements without a setting. But i've added a new `input` element that doesn't have it, so we need to check this here. Makes me think we shouldn't attach the event listener in the first place, letb me change that!

Comment (Skn0tt) [reply src/settingsView.ts]: adding it to the css selector is even smoother I think: https://github.com/microsoft/playwright-vscode/pull/569/commits/a1e1a8258eba61aaaa48e9fbafccd84b6eedc9e6

**Review by agg23 — APPROVED** [review]

**Pull request merged**